include/etiss/SimpleMemSystem.h

```diff
@@ -118,11 +119,30 @@ class MemSegment
         static std::default_random_engine generator{ static_cast<uint64_t>(0) };
         std::uniform_int_distribution<int> random_char_{ 0, 255 };
 
-        if (initString == "")
+        if (initString.find("0x") == 0)
         {
+            initString.erase(initString.begin(),initString.begin()+2);
+            std::stringstream mem_msg;
             for (etiss::uint64 i = 0; i < size_; ++i)
             {
-                mem_[i] = random_char_(generator);
+                const char* dataPtr = initString.substr(i%initString.length(),1).c_str();
+                char** endPtr = nullptr;
+                errno = 0;
+                uint8_t hexVal = static_cast<uint8_t>(strtol(dataPtr, endPtr, 16));
+                mem_[i] =  hexVal;
+
+                if (errno != 0){
```

Comment:
Thank you for clarifying. I was a bit confused after the  errno=EINVAL option was not triggered with a false "hex val string": e.g `0xDEADBEEFh ->  0xDEADBEEF0` instead of leading to an errno!=0.
A weird thing though is that vscode leads me to this linux header: 
https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/errno-base.h#L26